### PR TITLE
[Option A] ref(*): move spec into a file with version

### PR DESCRIPTION
**Pull request by michelleN**
* moves spec into SPEC.md with its own version
* also moves api specs into their own directory
* creates historical dir under each api dir
* creates working-draft dir under each api dir
* adds information about API group, version, and compatibility with other APIs to top of API spec docs
* Adds appendix with relevant sections around process and terminology.

Note: Some sections are not filled out and some links may be broken. Would appreciate some feedback before moving forward on whether this is the right approach or not.

### CODE REVIEW

File: apis/traffic-access/v1alpha1/traffic-target.md [hunk 1]
```diff
@@ -1,40 +1,44 @@
...
+define what traffic for specific protocols would look like. The kind can be
+different depending on what traffic a target is serving. In the following
+examples, `HTTPRouteGroup` is used for applications serving HTTP based traffic.
 
 To understand how this all fits together, first define the routes for some
 traffic.
 
 ```yaml
-apiVersion: v1alpha1
+apiVersion: v1alpha2
```
Comment (stefanprodan) [inline apis/traffic-access/v1alpha1/traffic-target.md:41]: I propose we remove the `apiVersion` from the examples and specify the version in the header. This way a backwards compatible change (such as adding an optional field) would not require a version bump in all examples.

Comment (grampelberg) [reply apis/traffic-access/v1alpha1/traffic-target.md:41]: Should we drop the entire header/metadata section?

Comment (michelleN) [reply apis/traffic-access/v1alpha1/traffic-target.md:41]: Yea I can see that would be a cleaner approach as well.

File: apis/traffic-split/v1alpha3/traffic-split.md [hunk 2]
```diff
@@ -1,9 +1,17 @@
...
-destinations.
+**API Group:** split.smi-spec.io
+
+**API Version:** v1alpha3
+
+**Compatible with:** specs.smi-spec.io/v1alpha2
+
+## Specification
+
+This specification defines the `TrafficTarget` resource which allows users
```
Comment (stefanprodan) [inline apis/traffic-split/v1alpha3/traffic-split.md:11]:
```suggestion
This specification defines the `TrafficSplit` resource which allows users
```

File: apis/traffic-split/traffic-split-wd.md [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+  then be required to resolve this link and reason about weights. By
+  making splits non-hierarchical, implementations become simpler and loose the
+  possibility of having circular references. It is still possible to build an
+  architecture that has nested split definitions, users would need to have a
+  secondary proxy to manage that.
+
+* TrafficSplits cannot be self-referential - consider the following definition:
+
+    ```yaml
+    apiVersion: split.smi-spec.io/v1alpha2
```
Comment (stefanprodan) [inline apis/traffic-split/traffic-split-wd.md:322]: please delete api version

File: apis/traffic-split/traffic-split-wd.md [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+
+## Example implementation
+
+This example implementation is included to illustrate how the `TrafficSplit`
+object operates. It is not intended to prescribe a particular implementation.
+
+Assume a `TrafficSplit` object that looks like:
+
+```yaml
+    apiVersion: split.smi-spec.io/v1alpha2
```
Comment (stefanprodan) [inline apis/traffic-split/traffic-split-wd.md:361]: please delete api version

File: apis/traffic-split/traffic-split-wd.md (hunk 3 above)
Comment (michelleN) [reply apis/traffic-split/traffic-split-wd.md]: good catch!

File: apis/traffic-split/traffic-split-wd.md (hunk 4 above)
Comment (michelleN) [reply apis/traffic-split/traffic-split-wd.md]: thnx

**Comment by michelleN** [comment]
Should I add the spec for the older versions of traffic specs and traffic split? Wondering what makes sense.

**Comment by stefanprodan** [comment]
@michelleN I think we should add all old versions according to #131

File: apis/traffic-metrics/traffic-metrics-WD.md [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+  version: v1alpha1
+```
+
+The default response, or requesting `/apis/metrics.smi-spec.io/v1alpha1/`
+would return:
+
+```yaml
+apiVersion: v1
+kind: APIResourceList
+groupVersion: metrics.smi-spec.io/v1alpha1
```
Comment (stefanprodan) [inline apis/traffic-metrics/traffic-metrics-WD.md:225]: I would search for `groupVersion` in all files and remove them 😄

**Pull request merged**